include/DataPoint.h

```diff
@@ -28,6 +28,13 @@
 
 namespace yap {
 
+/// precompiler flag for data storage type
+#ifdef DATA_POINT_FLOAT
```

Comment:
`DATA_POINT_FLOAT` -> `YAP_DATA_POINT_FLOAT`

And we should make this a member typedef of `DataPoint` simply called `type`